hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityConverter.java

```diff
@@ -58,18 +63,25 @@
   }
 
   Single<EntityResultSet> convert(
-      EntityRequest request, EntitiesResponse response, BaselineEntitiesResponse baselineResponse) {
+      EntityRequest request,
+      EntitiesResponse response,
+      BaselineEntitiesResponse baselineResponse,
+      Optional<GetLabelsResponse> labelsResponse) {
     return this.edgeLookupConverter
         .convert(request, response)
-        .flatMap(edgeLookup -> this.convert(request, response, baselineResponse, edgeLookup));
+        .flatMap(
+            edgeLookup ->
+                this.convert(request, response, baselineResponse, edgeLookup, labelsResponse));
   }
 
   private Single<EntityResultSet> convert(
       EntityRequest request,
       EntitiesResponse response,
       BaselineEntitiesResponse baselineResponse,
-      EdgeLookup edgeLookup) {
+      EdgeLookup edgeLookup,
+      Optional<GetLabelsResponse> labelsResponse) {
     Map<String, BaselineEntity> baselineEntityMap = getBaselineEntityMap(baselineResponse);
+    Map<String, String> labelsMap = getLabelsMap(labelsResponse);
```

Comment:
nit: can map optional labels response to empty map directly here, and pass the non optional value directly to `getLabelsMap`

Reply:
Done